account_reconcile_pos_payments/models/account_bank_statement.py

```diff
@@ -73,62 +72,73 @@ def _reconcile_pos(self):
             if not rec.journal_id.cb_child_ids:
                 raise UserError(_(
                     'The journal %s has no CB journals.'
-                    ) % rec.journal_id.display_name)
+                ) % rec.journal_id.display_name)
 
             if not rec.journal_id.cb_contract_number:
                 raise UserError(_(
                     'The journal %s has no CB Contract number.'
-                    ) % rec.journal_id.display_name)
+                ) % rec.journal_id.display_name)
 
             # Process lines that match the pattern
             line_ids = rec.line_ids.search([
                 ('statement_id', '=', rec.id),
                 ('journal_entry_ids', '=', False),
-                ('name', 'ilike', '%%%s%%' % rec.journal_id.cb_contract_number),
+                ('name', 'ilike',
+                    '%%%s%%' % rec.journal_id.cb_contract_number),
             ])
 
             i = 0
             for line in line_ids:
                 i += 1
-                _logger.debug('Processing line %s (%d/%d)' % (line.name, i, len(line_ids)))
+                _logger.debug('Processing line %s (%d/%d)' %
+                              (line.name, i, len(line_ids)))
 
                 # Try to find a CB statement older than our statement line
                 # but not older than 3 days before
-                limit_date = (
-                    datetime.strptime(line.date, '%Y-%m-%d')
-                    - timedelta(days=rec.journal_id.cb_delta_days))
+                limit_date = (datetime.strptime(
+                    str(line.date), '%Y-%m-%d') -
+                    timedelta(days=rec.journal_id.cb_delta_days))
```

Comment:
```suggestion
                limit_date = line.date - timedelta(days=rec.journal_id.cb_delta_days)
```

in `12.0` we don't need to convert date fields to datetime objects, so this should work

Can you do the same with the other cases (`strftime` / `strptime`)?